File: solutions/python/question2.py
"""Q2 — Reservas de coworking.

Abra questions/q2_room_reservations/LEVEL_1.md e preserve as assinaturas públicas.
"""


class RoomReservationService:
    def add_room(self, room_id: str, capacity: int) -> bool:
        return False

    def book(
        self,
        timestamp: int,
        booking_id: str,
        room_id: str,
        user_id: str,
        start: int,
        end: int,
        attendees: int,
    ) -> bool:
        return False

    def cancel(self, timestamp: int, booking_id: str) -> bool:
        return False

    def room_schedule(self, timestamp: int, room_id: str) -> list[str]:
        return []

    def available_rooms(
        self, timestamp: int, start: int, end: int, attendees: int
    ) -> list[str]:
        return []

    def top_users(self, timestamp: int, n: int) -> list[str]:
        return []

    def join_waitlist(
        self,
        timestamp: int,
        request_id: str,
        room_id: str,
        user_id: str,
        start: int,
        end: int,
        attendees: int,
    ) -> bool:
        return False

    def cancel_waitlist(self, timestamp: int, request_id: str) -> bool:
        return False

    def get_waitlist(self, timestamp: int, room_id: str) -> list[str]:
        return []

    def book_series(
        self,
        timestamp: int,
        series_id: str,
        room_id: str,
        user_id: str,
        starts: list[int],
        duration: int,
        attendees: int,
    ) -> bool:
        return False

    def cancel_series(self, timestamp: int, series_id: str) -> int:
        return 0

File: solutions/typescript/question2.ts
/**
 * Q2 — Reservas de coworking.
 *
 * Abra questions/q2_room_reservations/LEVEL_1.md e implemente somente o nível
 * atualmente desbloqueado. Preserve todas as assinaturas públicas.
 */
export class RoomReservationService {
  addRoom(roomId: string, capacity: number): boolean {
    return false;
  }

  book(
    timestamp: number,
    bookingId: string,
    roomId: string,
    userId: string,
    start: number,
    end: number,
    attendees: number
  ): boolean {
    return false;
  }

  cancel(timestamp: number, bookingId: string): boolean {
    return false;
  }

  roomSchedule(timestamp: number, roomId: string): string[] {
    return [];
  }

  availableRooms(timestamp: number, start: number, end: number, attendees: number): string[] {
    return [];
  }

  topUsers(timestamp: number, n: number): string[] {
    return [];
  }

  joinWaitlist(
    timestamp: number,
    requestId: string,
    roomId: string,
    userId: string,
    start: number,
    end: number,
    attendees: number
  ): boolean {
    return false;
  }

  cancelWaitlist(timestamp: number, requestId: string): boolean {
    return false;
  }

  getWaitlist(timestamp: number, roomId: string): string[] {
    return [];
  }

  bookSeries(
    timestamp: number,
    seriesId: string,
    roomId: string,
    userId: string,
    starts: number[],
    duration: number,
    attendees: number
  ): boolean {
    return false;
  }

  cancelSeries(timestamp: number, seriesId: string): number {
    return 0;
  }
}
